shared/src/metabase/types.cljc

```diff
@@ -251,29 +251,29 @@
   ;; that's an intentional mapping or not. But it does mean that lots of extra columns will be offered a conversion
   ;; (think Price being offerred to be interpreted as a date)
   (let [numeric-types [:type/BigInteger :type/Integer :type/Decimal]]
-    (reduce #(assoc %1 %2 {:Coercion/UNIXMicroSeconds->DateTime :type/DateTime
-                           :Coercion/UNIXMilliSeconds->DateTime :type/DateTime
-                           :Coercion/UNIXSeconds->DateTime      :type/DateTime})
+    (reduce #(assoc %1 %2 {:Coercion/UNIXMicroSeconds->DateTime :type/Instant
+                           :Coercion/UNIXMilliSeconds->DateTime :type/Instant
+                           :Coercion/UNIXSeconds->DateTime      :type/Instant})
             {:type/Text   {:Coercion/ISO8601->Date     :type/Date
                            :Coercion/ISO8601->DateTime :type/DateTime
                            :Coercion/ISO8601->Time     :type/Time}}
             numeric-types)))
 
-(defn ^:export is_coerceable
+(defn ^:export is-coerceable
   "Returns a boolean of whether a field base-type has any coercion strategies available."
   [base-type]
   (boolean (contains? coercions (keyword base-type))))
 
-(defn ^:export effective_type_for_coercion
+(defn ^:export effective-type-for-coercion
   "The effective type resulting from a coercion."
   [coercion]
   ;;todo: unify this with the coercions map above
-  (get {:Coercion/ISO8601->Date :type/Date
-        :Coercion/ISO8601->DateTime :type/DateTime
-        :Coercion/ISO8601->Time :type/Time
-        :Coercion/UNIXMicroSeconds->DateTime :type/DateTime
-        :Coercion/UNIXMilliSeconds->DateTime :type/DateTime
-        :Coercion/UNIXSeconds->DateTime :type/DateTime}
+  (get {:Coercion/ISO8601->Date              :type/Date
+        :Coercion/ISO8601->DateTime          :type/DateTime
+        :Coercion/ISO8601->Time              :type/Time
+        :Coercion/UNIXMicroSeconds->DateTime :type/Instant
+        :Coercion/UNIXMilliSeconds->DateTime :type/Instant
+        :Coercion/UNIXSeconds->DateTime      :type/Instant}
```

Comment:
`:Coercion/UNIXMilliSeconds->DateTime :type/Instant` should the coercion be renamed as well? Possibly to the generic `:Coercion/UNIXMilliSeconds->Temporal` or `->Instant`?

Reply:
Arguably, `->Instant` _would_ be more specific here, but `->DateTime` isn't wrong. It *is* something with a date and a time, so it *is* a DateTime. I didn't want to change a millions things TBH

Reply:
agreed. when i first read the description of this PR i assumed it would remain a `:type/DateTime` and just be overridden in the big query query processor. I was surprised to see it go everywhere. I still haven't thought it through but that's where my initial expectation was.

Reply:
I could have just wrote a workaround in the BigQuery QP but I thought that would get us in to trouble in the future when we wanted to support user-defined coercions in the future. The BigQuery QP code does the right thing if it has specific-enough type info, so it seemed like a better long-term fix to make the info more specific everywhere so BigQuery could make use of it rather than write a workaround that would only handle these known types

Reply:
yeah i'm surprised it wasn't more localized to this:
```clojure
(doseq [[unix-timestamp-type bigquery-fn] {:seconds      :timestamp_seconds
                                           :milliseconds :timestamp_millis
                                           :microseconds :timestamp_micros}]
  (defmethod sql.qp/unix-timestamp->honeysql [:bigquery unix-timestamp-type]
    [_ _ expr]
    (with-temporal-type (hsql/call bigquery-fn expr) :timestamp)))
```